### PR TITLE
Improve error if numpy is requested but missing

### PR DESCRIPTION
**Pull request by rec**

### CODE REVIEW

File: bibliopixel/main/common_flags.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def add_project_flags(parser):
         help='Number of pixels controlled by the driver.')
 
     parser.add_argument(
-        '--numpy_dtype', default=None, help='Use this numpy datatype.')
+        '--numpy_dtype', '-m', default=None, help='Use this numpy datatype.')
```
Comment (adammhaile) [inline bibliopixel/main/common_flags.py]: -m ? not -n?

**Comment by rec** [comment]
We use `-n` already for "number of pixels".

Actually, now we have `--dimensions`/`--dim`, that sort of takes the place of `-n`, so I could obsolete it...

**Comment by adammhaile** [comment]
Yeah, that's better.

**Comment by rec** [comment]
Done!

**Comment by rec** [comment]
Ah, I pushed an extra commit onto the end of this sequence which makes life much simpler for the user - no more obscure "numpy_dtype".

Now you can say things like:

    bp lites.json
    bp lites.json --numbers=python
    bp lites.json -n python  
    # Those last three are all the same.

    bp lites.json -n int
    bp lites.json -n float
    bp lites.json -n uint8

or in your project:

    {"numbers": "float"},

or in your animation class:

    class MyAnimation(BaseAnimation):
        PROJECT = {"numbers": "float"}

**Comment by adammhaile** [comment]
Very cool. +1

**Pull request merged**